src/Symfony/Component/DependencyInjection/Compiler/ResolveNamedArgumentsPass.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * Resolves named arguments to their corresponding numeric index.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ResolveNamedArgumentsPass extends AbstractRecursivePass implements CompilerPassInterface
+{
+    private $constructors = array();
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $throwingAutoloader = function ($class) { throw new \ReflectionException(sprintf('Class %s does not exist', $class)); };
+        spl_autoload_register($throwingAutoloader);
+
+        try {
+            parent::process($container);
+        } finally {
+            spl_autoload_unregister($throwingAutoloader);
+
+            // Free memory
+            $this->constructors = array();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function processValue($value, $isRoot = false)
+    {
+        if (!$value instanceof Definition) {
+            return parent::processValue($value, $isRoot);
+        }
+
+        $class = $value->getClass();
+        $originalArguments = $arguments = $value->getArguments();
+
+        foreach ($arguments as $key => $argument) {
+            if (!is_string($key)) {
```

Comment:
Should be `!is_string($key) || '$' !== $key[0]`, right? Given `<argument index="0">foo</argument>` would reach the exception below